api/handlers_mobile.go

```diff
@@ -784,3 +793,91 @@ func MobileSearchHandler(c *gin.Context) {
 		NewInternalError(err).Abort(c)
 	}
 }
+
+// call feed api to get results like search - https://kabbalahmedia.info/feed_api/feed
+// write sql to get
+func MobileFeed(c *gin.Context){
+	// get input json
+	var r MobileFeedRequest
+	if c.Bind(&r) != nil {
+		return
+	}
+
+	feedInputJson, err := json.Marshal(r)
+	if err != nil {
+		NewInternalError(err).Abort(c)
+	}
+
+	feedApi, err := getFeedApi("feed")
+	if err != nil {
+		NewInternalError(err).Abort(c)
+	}
+
+	feedResponseObj, err := http.Post(feedApi, "application/json", strings.NewReader(string(feedInputJson)))
+	if err != nil {
+		NewInternalError(err).Abort(c)
+	}
+
+	// convert response body to byte arr and then get the feed array
+	feedRespBytes, err := io.ReadAll(feedResponseObj.Body)
+	if err != nil {
+		log.Error(err.Error())
+	}
+
+	feedBody := new(feedResponseType)
+	if err = json.Unmarshal(feedRespBytes, feedBody); err != nil || feedBody == nil {
+		log.Error(err.Error())
+	}
+
+	imagesUrlTemplate := viper.GetString("content_unit_images.url_template")
+	var mobilefeedResponse []*MobileFeedResponseItem
+
+	db := c.MustGet("MDB_DB").(*sql.DB)
+	language := c.Query("language");
+
+	var cuIds []string
+	itemsMap := make(map[string]*MobileFeedResponseItem)
+
+	for _, item := range feedBody.Feed {
+		imageStr := fmt.Sprintf(imagesUrlTemplate, item.ContentUnitUid)
+
+		feedResp := &MobileFeedResponseItem {
+			ContentUnitUid: item.ContentUnitUid,
+			Type: item.ContentType,
+			Image: &imageStr,
+			Date: item.Date,
+			// Views: ,
+			Title: getContentUnitName(item.ContentUnitUid, language, db),
+		}
+
+		itemsMap[item.ContentUnitUid] = feedResp
+		cuIds = append(cuIds, item.ContentUnitUid)
+		mobilefeedResponse = append(mobilefeedResponse, feedResp)
+	}
+
+	// set views
+	mapViewsToMobileResponseItems[*MobileFeedResponseItem](cuIds, itemsMap)
+
+	c.JSON(http.StatusOK, mobilefeedResponse)
+}
+
+func getContentUnitName(uid string, language string, db *sql.DB) string {
+	QUERY_TITLE := `SELECT name
```

Comment:
better use entity manager (sql boiler) and not queries.Raw (more security)

Reply:
Do you have an example for this, what exactly you mean?

Reply:
units, err := mdbmodels.ContentUnits(
		qm.WhereIn("id IN ?", utils.ConvertArgsInt64(cuIDs)...),
		qm.Load("CollectionsContentUnits"),
		qm.Load("CollectionsContentUnits.Collection")).
		All(db)
	if err != nil {
		return nil, nil, NewInternalError(err)
	}